ext/serializer.c

```diff
@@ -1799,6 +1852,10 @@ void ddtrace_serialize_span_to_array(ddtrace_span_data *span, zval *array) {
         smart_str_free(&metrics_str);
     })
 
+    if (inferred_span) {
+        ddtrace_serialize_span_to_array(inferred_span, array);
```

Comment:
I would prefer if we could do all the transfer_data calls here (from serialized to inferred serialized), after the serializing step. But I won't insist.